src/priority.js

```diff
@@ -0,0 +1,81 @@
+'use strict'
+
+class PriorityNode {
+  constructor(data, priority) {
+    this.data = data;
+    this.next = null;
+    this.priority = priority;
+  }
+}
+
+export default class Priority {
+
+  constructor(data) {
+    this.head = null
+    this.tail = null
+    this.size = 0
+  }
+
+  enqueue( data, priority ) {
+    const newPriorityNode = new PriorityNode( data, priority ) //{ data: 'foo', next: null}
+    if( this.size === 0 ) {
+      this.head = newPriorityNode
+      this.tail = newPriorityNode
+      this.size++
+    } else {
+      newPriorityNode.next = this.tail // newPriorityNode = { data: 'foo', next:{data: 'roo', next: null}}
+      //this.tail = {{data: 'roo', next: null}
```

Comment:
I tend to not like comments in code. Your code should be written well enough that I can figure out what it is doing without comments. 
Try to have your code show rather than you haveing to tell.
This can be done with clean logical, well formatted, meaningfully written code. 
Your code here looks fine so the comments make it look cluttered and less neat.